src/scenes/Lines/scenes/Editor/EditorFrame.tsx

```diff
@@ -20,17 +23,53 @@ import PageHeader from 'components/PageHeader';
 import NavigateConfirmBox from 'components/ConfirmNavigationDialog';
 import Loading from 'components/Loading';
 import { setSavedChanges } from 'actions/editor';
+import { loadNetworks, saveNetwork } from 'actions/networks';
+import { Network } from 'model/Network';
+import Provider from 'model/Provider';
+import { PrimaryButton } from '@entur/button';
+import ConfirmDialog from 'components/ConfirmDialog';
+
+const findNetworkIdByProvider = (
+  provider: Provider,
+  networks: Network[]
+): string | undefined =>
+  networks.find(
+    (network) =>
+      network.id?.split(':')?.[0]?.toUpperCase() === provider.codespace?.xmlns
+  )?.id;
+
+const createAndGetNetwork = (
+  dispatch: any,
+  authorityRef: string,
+  activeProvider: Provider
+): Promise<string> =>
+  dispatch(
+    saveNetwork(
+      {
+        name: activeProvider.codespace?.xmlns,
+        authorityRef: authorityRef,
+      },
+      false
+    )
+  )
+    .then(() => dispatch(loadNetworks()))
+    .then((newNetworks: Network[]) =>
+      findNetworkIdByProvider(activeProvider, newNetworks)
+    );
 
 const EditorFrame = (props: RouteComponentProps<MatchParams>) => {
   const [flexibleLine, setFlexibleLine] = useState<FlexibleLine>({});
   const [showConfirm, setShowConfirm] = useState<boolean>(false);
 
   const { formatMessage } = useSelector(selectIntl);
-  const dispatch = useDispatch();
-  const { flexibleLines, organisations, networks, editor } = useSelector<
-    GlobalState,
-    GlobalState
-  >((s) => s);
+  const dispatch = useDispatch<any>();
```

Comment:
Går det att få type på dispatchen? 👼

Reply:
Den er litt seig. Siden den er async. Vet ikke helt hva som skal inn for å være ærlig. Vi. har sånn useDispatch<any> på alle async-kall så kanskje vi skal lage en task på det? :)